lib/tags/gomodifytags.js

```diff
@@ -0,0 +1,176 @@
+        icon: 'tag',
+        detail: 'Modifying tags only works with a single cursor'
+      })
+      return
+    }
+
+    return this.goconfig.locator.findTool('gomodifytags').then((cmd) => {
+      const dialog = new TagsDialog({
+        mode: mode
+      })
+      dialog.onAccept = (options) => this.modifyTags(editor, options, mode, cmd)
+      dialog.attach()
+    })
+  }
+
+  buildArgs (editor, options, mode) {
+    const {tags, useSnakeCase, sortTags} = options
+
+    // if there is a selection, use the -line flag,
+    // otherwise just use the cursor offset (and apply modifications to entire struct)
+    const args = ['-file', editor.getTitle()]
+    const selection = editor.getSelectedBufferRange()
+    if (selection && !selection.start.isEqual(selection.end)) {
+      args.push('-line')
+      if (selection.isSingleLine()) {
+        args.push(`${selection.start.row + 1}`)
+      } else {
+        args.push(`${selection.start.row + 1},${selection.end.row + 1}`)
+      }
+    } else {
+      args.push('-offset')
+      args.push(this.editorByteOffset(editor))
+    }
+
+    if (editor.isModified()) {
+      args.push('-modified')
+    }
+
+    if (!useSnakeCase) {
+      args.push('-transform')
+      args.push('camelcase')
+    }
+    if (sortTags) {
+      args.push('-sort')
+    }
+
+    if (mode === 'Add') {
+      const tagNames = []
+      const opts = []
+      for (const t of tags) {
+        tagNames.push(t.tag)
+        if (t.option && t.option.length) {
+          opts.push(`${t.tag}=${t.option}`)
+        }
+      }
+      if (opts.length > 0) {
+        args.push('-add-options')
+        args.push(opts.join(','))
+      }
+      if (tagNames.length === 0) {
+        tagNames.push('json')
+      }
+      args.push('-add-tags', tagNames.join(','))
+    } else if (mode === 'Remove') {
+      const tagNames = []
+      const opts = []
+      if (!tags || !tags.length) {
+        args.push('-clear-tags')
+      } else {
+        for (const t of tags) {
+          if (t.option && t.option.length) {
+            opts.push(`${t.tag}=${t.option}`)
+          } else {
+            tagNames.push(t.tag)
+          }
+        }
+        if (tagNames.length > 0) {
+          args.push('-remove-tags')
+          args.push(tagNames.join(','))
+        }
+        if (opts.length > 0) {
+          args.push('-remove-options')
+          args.push(opts.join(','))
+        }
+      }
+    }
+    return args
+  }
+
+  modifyTags (editor, options, mode, cmd) {
+    const executorOptions = this.goconfig.executor.getOptions('file')
+
+    if (editor.isModified()) {
+      let archive = ''
+      archive += editor.getTitle() + '\n'
+      archive += Buffer.byteLength(editor.getText(), 'utf8') + '\n'
+      archive += editor.getText()
+
+      executorOptions.input = archive
+    }
+
+    const args = this.buildArgs(editor, options, mode)
+
+    if (atom.inDevMode()) {
+      console.log('(go-plus): executing: gomodifytags ' + args.join(' '))
```

Comment:
We can remove it if you'd like.  I actually was thinking it would be nice if there was an option on the executor just to log the commands it executes.  I found it very helpful for troubleshooting when you want to run the commands outside of Atom and look at results.

Is there a similar check on the `atom` global that can tell me if I'm running specs?